IPCLib/Socket.cpp

```diff
@@ -3,15 +3,38 @@
 #include <sstream>
 #include <cassert>
 #include <thread>
+#include <mutex>
 
 /// @brief Checks if the server is still open
 #define CHECK_OPEN() \
     if (!Open) return IPCLIB_CLOSED_CONNECTION_ERROR;
 
+#define ERROR_STATE_INV         0b11101111
+#define ERROR_STATE             0b00010000
+#define STARTED_RECEIVING_STATE 0b00001000
+#define RECEIVED_STATE          0b00000100
+#define RECEIVING_STATE         0b00000010
+#define STOP_STATE              0b00000001
+
+#define GET_ERROR()             ((m_state & ERROR_STATE) > 0)
+#define GET_STARTED_RECEIVING() ((m_state & STARTED_RECEIVING_STATE) > 0)
+#define GET_RECEIVED()          ((m_state & RECEIVED_STATE) > 0)
+#define GET_RECEIVING()         ((m_state & RECEIVING_STATE) > 0)
+#define GET_STOP()              ((m_state & STOP_STATE) > 0)
```

Comment:
Should this be != instead of = so it works on signed ints?

Reply:
what? m_state is an unsinged int? there is no = here?